### ISSUE TITLE
Add Option to use different Spam-Score-Headers

### ISSUE THREAD

**Issue by nyarlatothep** [friedPotat0/Spam-Scores#5]
First of all, terrific work for bringing this addon to Thunderbird 78 users! Thanks a lot!

I am an old user of the plugin "Rspamd-spamness" which didn't make it to the new Thunderbird version.
It would be great, if your plugin would parse more and different Mail-Headers. It now seems to work with "X-Spam-Status", but i'm using different Mail-Servers and would appreciate parsing something like:

X-MyCompany-MailScanner-SpamCheck: spam, SpamAssassin (not cached, score=8, required 3,....)
X-MailScanner-SpamCheck: spam, SpamAssassin (not cached, score=5, required 3,....)
...

Would be nice for future versions, if there could be a option for adding different headers.

Keep up the good work.
Regards,
nyar

**Comment by friedPotat0** [comment]
Thanks for your feedback!

It should be no problem to add further support for this mail headers. Are there any detailed rules listed in this mail headers that can be parsed or only the overall score?

It would be nice if you could provide a full header example for each of your cases.

At the moment it is not planed to add an add-on setting to adjust the score parsing for custom mail headers directly by the user. So if there are any additional cases to consider, please feel free to post them here.

**Comment by nyarlatothep** [comment]
Hi.

Mostly i am using MailScanner as the spamfilter, which produces something like this

```
X-MyCompany-MailScanner-From: sales@xxx.com
X-MyCompany-MailScanner-SpamScore: ssssssss
X-MyCompany-MailScanner-SpamCheck: spam, SpamAssassin (not cached, score=8.609,
	required 3, BAYES_50 0.80, FREEMAIL_FORGED_FROMDOMAIN 0.25,
	FREEMAIL_FROM 0.00, FROM_NOT_REPLYTO 2.00, GB_FREEMAIL_DISPTO 0.00,
	HEADER_FROM_DIFFERENT_DOMAINS 0.25, HTML_FONT_LOW_CONTRAST 0.00,
	HTML_MESSAGE 0.00, KAS_SPAM_BRBL 1.00, RDNS_NONE 0.79,
	SCHAALIT_HEADER_4074 1.50, SPF_HELO_NONE 0.00, SPF_NONE 0.00,
	SPOOFED_FREEMAIL 2.00, SPOOFED_FREEMAIL_NO_RDNS 0.02)
```

The name of the header differs from different MailScanner-Installations, so i can't mention all of them (for examply "X-SecondCompany-MailScanner-SpamCheck", etc....). All of them should have the keyword "MailScanner" in their name.

It can happen, that there are both my MailScanner-Headers and "X-Spam-Status" from somewhere else in the same mail, don't know how to tell your plugin which one is the preferred to show.

Hope this helps.

Thanks again,
nyar

**Comment by friedPotat0** [comment]
Ok. So there are more things to do than simply adding support for MailScanner headers. Is it correct that after "X-MyCompany-MailScanner-SpamScore:" the total score (8.609) is written?

I've already implemented the MailScanner headers with variable company/domain names (X-*-MailScanner-SpamCheck/SpamScore). It is tested based on the example you've given.

At the moment different spam headers in a single mail are parsed in the following order (first match is displayed, others are ignored):
1. x-spamd-result
2. x-spam-score
3. x-spam-status
4. x-*-mailscanner-spamscore

If you like to test the new version, you can download the files from the mailscanner-headers branch (https://github.com/friedPotat0/Spam-Scores/archive/mailscanner-headers.zip) and create a new ZIP file containing all files in the "Spam-Scores-mailscanner-headers" folder. Then you can add the file to thunderbird by drag & drop to install the new version.

**Comment by nyarlatothep** [comment]
Wow, fast work!

I installed you new version, but it does work not as expected.

It shows no numeric score in the mail list view and it shows "Spam Score: sssssss" in the action bar (and all triggered rules).
So it seems to use the "x-*-mailscanner-**spamscore**" header, which displays the score not as a number but with the letter "s" for each score-point. (The letter "s" is configurable by the MailScanner-user, so you can't depend on it). This is for better usage within sieve-filtering.

It is correct, that in the header "X-MyCompany-MailScanner-**SpamCheck**:" the total score is written as a number. You should parse in this header for the keyword **score=8.609**.

So perhaps you should change your code from "....SpamScore" to ".....SpamCheck".

Thanks for your time
nyar

**Comment by friedPotat0** [comment]
Thanks for clarifiying. I was not aware of the fact that such a configurable spamscore header exists. I've changed the header to spamcheck. Please check out the new version (https://github.com/friedPotat0/Spam-Scores/archive/mailscanner-headers.zip).

**Comment by nyarlatothep** [comment]
Almost there.... :)

The mail list view now shows "NaN" instead of the number, but in the action bar the number is correctly displayed.
Maybe that's because the number from the Mailscanner-score is with three digits (8.609) behind the point instead the Spamassassin-score with only one digit (would be 8.6)....?

Still my problem exists with the order of parsing, because if a Spamassassin header is present it is shown instead my own Mailscanner-header, as you mentioned in your posting above. But i guess that's a little more complicated to change, so i am happy with all your support right now.

Really great work.
nyar

**Comment by friedPotat0** [comment]
Have you tried restarting Thunderbird and repairing the folder containing the mails with this spam header? That is only necessary for the spam column in the mail list view.

**Comment by friedPotat0** [comment]
I've opened a new issue for the header priority option. #7

**Comment by nyarlatothep** [comment]
> 
> 
> Have you tried restarting Thunderbird and repairing the folder containing the mails with this spam header? That is only necessary for the spam column in the mail list view.

Both done, not getting better, still showing "NaN" in spam column, but correct Spam-Score in message action bar.

**Comment by friedPotat0** [comment]
Hopefully it should be fixed now... :slightly_smiling_face:

**Comment by nyarlatothep** [comment]
> 
> 
> Hopefully it should be fixed now... 🙂

Correct. Thanks for the fast and great work!

**Comment by friedPotat0** [comment]
The change is now publicly available in the new version 1.1.3
